dimod/responses/template_response.py

```diff
@@ -0,0 +1,561 @@
+            increasing energy.
+
+        Examples:
+            >>> response = TemplateResponse()
+            >>> response.add_sample({0: -1}, 1, data={'n': 5})
+            >>> response.add_sample({0: 1}, -1, data={'n': 1})
+            >>> list(response.items())
+            [({0: 1}, -1), ({0: -1}, 1)]
+            >>> list(response.items(data=True))
+            [({0: 1}, -1, {'n': 1}), ({0: -1}, 1, {'n': 5})]
+
+        Note:
+            Depreciation Warning: This method of access is being depreciated.
+            it can be replaced by `response.data(keys=['sample', 'energy'])`
+
+        """
+        warnings.warn("'items' will be depreciated in dimod 1.0.0", DeprecationWarning)
+        if data:
+            return zip(self.samples(), self.energies(), self.data())
+        return zip(self.samples(), self.energies())
+
+    def add_sample(self, sample, energy, num_occurences=1, **kwargs):
+        """Loads a sample and associated energy into the response.
+
+        Args:
+            sample (dict): A sample as would be returned by a discrete
+                model solver. Should be a dict of the form
+                {var: value, ...}.
+            energy (number): The energy associated with the given
+                sample.
+            num_occurences (int): The number of times the sample occurred.
+            **kwargs
+
+        Examples:
+            >>> response = dimod.TemplateResponse()
+            >>> response.add_sample({0: -1}, 1)
+            >>> response.add_sample({0: 1}, -1, sample_idx=1)
+            >>> list(response.data())
+            [{'energy': -1, 'num_occurences': 1, 'sample': {0: 1}, 'sample_idx': 1},
+             {'energy': 1, 'num_occurences': 1, 'sample': {0: -1}}]
+
+        """
+
+        kwargs['sample'] = sample
+        kwargs['energy'] = energy
+        kwargs['num_occurences'] = num_occurences
+
+        self.add_data_from((kwargs,))
+
+    def add_samples_from(self, samples, energies, num_occurences=1, **kwargs):
+        """Loads samples and associated energies from iterators.
+
+        Args:
+            samples (iterator): An iterable object that yields
+                samples. Each sample should be a dict of the form
+                {var: value, ...}.
+            energies (iterator): An iterable object that yields
+                energies associated with each sample.
+            num_occurences (int): Default 1. The number of times the sample
+                occurred. This is applied to each sample.
+            **kwargs
+
+
+        Examples:
+            >>> response = dimod.TemplateResponse()
+            >>> response.add_samples_from([{0: -1}, {0: 1}], [1, -1], dataval='test')
+            >>> list(response.data())
+            [{'dataval': 'test', 'energy': -1, 'num_occurences': 1, 'sample': {0: 1}},
+             {'dataval': 'test', 'energy': 1, 'num_occurences': 1, 'sample': {0: -1}}]
+
+        """
+        stop = object()  # signals that the two iterables are different lengths
```

Comment:
Why not `None`?

Reply:
object has a unique ID, but None does not.